### PR TITLE
Mqx 165 create date filter

### PR DESCRIPTION
**Pull request by skandarhayek**

### CODE REVIEW

File: src/framework/filterControl/forms/DateRangeFilterForm.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    const filter = new Filter(
+      this.props.filterOption,
+      dateRange
+    );
+    this.props.onAddFilter(filter);
+  }
+
+  onKeyUp(event) {
+    if (event.key !== 'Enter') return;
+    this.onClickAddButton();
```
Comment (mariacasancho) [inline src/framework/filterControl/forms/DateRangeFilterForm.jsx]: why not check for enter and do the function call?

File: src/framework/filterControl/forms/DateRangeFilterForm.jsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  constructor(props) {
+    super(props);
+
+    this.onChange = this.onChange.bind(this);
+    this.onClickAddButton = this.onClickAddButton.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+
+    let dateRange;
+    if (this.props.comparisonValue === undefined) {
```
Comment (mariacasancho) [inline src/framework/filterControl/forms/DateRangeFilterForm.jsx]: if you want to check for undefined you should use type of. 

You can also just do:

if (this.props.comparisonValue) { 
dateRange = {
        startDate: moment(this.props.comparisonValue.startDate),
        endDate: moment(this.props.comparisonValue.endDate),
      };
......

and you check more stuff than just undefined

File: src/framework/filterControl/forms/DateRangeFilterForm.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
+
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { DateRange as ReactDateRange } from 'react-date-range';
+import React, {
+  Component,
+} from 'react';
```
Comment (arijitdasgupta) [inline src/framework/filterControl/forms/DateRangeFilterForm.jsx]: Maybe let's put React import before all this, no biggies, but like every other file.

**Comment by mariacasancho** [comment]
when I select some rage date, click update results and then click again in the filter, the date selected is not the one that I selected before

File: src/framework/filterControl/forms/DateRangeFilterForm.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      endDate: this.state.dateRange.endDate.toDate(),
+    };
+    const filter = new Filter(
+      this.props.filterOption,
+      dateRange
+    );
+    this.props.onAddFilter(filter);
+  }
+
+  onKeyUp(event) {
```
Comment (arijitdasgupta) [inline src/framework/filterControl/forms/DateRangeFilterForm.jsx]: Do we really need this?

Also it's not bound to an event emitter, so the function is not getting called either way.

**Comment by skandarhayek** [comment]
> when I select some rage date, click update results and then click again in the filter, the date selected is not the one that I selected before

Fixed

File: src/framework/services/filter/ComparisonTypes.js [hunk 5]
```diff
@@ -4,4 +4,5 @@ export default {
   MAX: 'max',
   MIN: 'min',
   ONE_OF: 'one of',
+  DATE_RANGE: 'date range',
```
Comment (mariacasancho) [inline src/framework/services/filter/ComparisonTypes.js]: order

File: src/framework/services/filter/Filter.spec.js [hunk 6]
```diff
@@ -58,6 +58,26 @@ describe('Filter', () => {
...
+        const filter = new Filter(filterOption, normalizedComparisonValues);
+        // The PhantomJS toLocaleDateString does not have the same default options as chrome
+        expect(filter.humanizeComparisonValue()).toBe(
+          `${
+            normalizedComparisonValues.startDate.toLocaleDateString('de-DE')
+          } - ${
+            normalizedComparisonValues.endDate.toLocaleDateString('de-DE')
+          }`
+        );
+        // .toBe('30.11.1989 - 30.11.2019');
```
Comment (mariacasancho) [inline src/framework/services/filter/Filter.spec.js]: after talking with alex, im ok with having this comment more clear next to the one talking about phantomJs or removing it

**Review by mariacasancho — COMMENTED** [review]
just few suggestions, else good job!

**Review by arijitdasgupta — APPROVED** [review]
Good job! Couple of stuff, but otherwise looks good.

**Comment by skandarhayek** [comment]
PRA @mariacasancho @arijitdasgupta

**Review by mariacasancho — APPROVED** [review]

**Pull request merged**